core/org/commcare/core/process/XmlFormRecordProcessor.java

```diff
@@ -26,22 +26,25 @@
  */
 public class XmlFormRecordProcessor {
 
-    public static void process(UserSandbox sandbox, InputStream stream) throws InvalidStructureException, IOException, XmlPullParserException, UnfullfilledRequirementsException, StorageFullException {
-        final UserSandbox mSandbox = sandbox;
-
-        InputStream is = stream;
-
-        DataModelPullParser parser = new DataModelPullParser(is, new TransactionParserFactory() {
+    public static void process(final UserSandbox sandbox, InputStream stream) throws InvalidStructureException, IOException, XmlPullParserException, UnfullfilledRequirementsException, StorageFullException {
```

Comment:
Adding the alternate constructor here is the fix, the rest of this code fixes the CLI against complex ICDS workflows